data-prepper-plugins/saas-source-plugins/jira-source/src/main/java/org/opensearch/dataprepper/plugins/source/jira/JiraService.java

```diff
@@ -0,0 +1,361 @@
+                        ((Map) issue.getFields().get(PROJECT)).get(NAME).toString());
+            }
+
+            long created = 0;
+            if (Objects.nonNull(issue.getFields()) && issue.getFields().get(CREATED)
+                    .toString().length() >= 23) {
+                String charSequence = issue.getFields().get(CREATED).toString().substring(0, 23) + "Z";
+                OffsetDateTime offsetDateTime = OffsetDateTime.parse(charSequence);
+                new Date(offsetDateTime.toInstant().toEpochMilli());
+                created = offsetDateTime.toEpochSecond() * 1000;
+            }
+            issueMetadata.put(CREATED, String.valueOf(created));
+
+            long updated = 0;
+            if (issue.getFields().get(UPDATED).toString().length() >= 23) {
+                String charSequence = issue.getFields().get(UPDATED).toString().substring(0, 23) + "Z";
+                OffsetDateTime offsetDateTime = OffsetDateTime.parse(charSequence);
+                new Date(offsetDateTime.toInstant().toEpochMilli());
+                updated = offsetDateTime.toEpochSecond() * 1000;
+            }
+            issueMetadata.put(UPDATED, String.valueOf(updated));
+
+            issueMetadata.put(ISSUE_KEY, issue.getKey());
+            issueMetadata.put(CONTENT_TYPE, JiraContentType.ISSUE.getType());
+            String id = _ISSUE + issueMetadata.get(PROJECT_KEY) + "-" + issue.getKey();
+
+            itemInfoQueue.add(createItemInfo(id, issueMetadata));
+
+            if (Objects.nonNull(issueMetadata.get(PROJECT_KEY)) && !jiraProjectCache
+                    .containsKey(issueMetadata.get(PROJECT_KEY))) {
+                jiraProjectCache.put((String) issueMetadata.get(PROJECT_KEY), LIVE);
+            }
+
+        });
+    }
+
+    /**
+     * Method to get Issues.
+     *
+     * @param jql           input parameter.
+     * @param startAt       the start at
+     * @param configuration input parameter.
+     * @return InputStream input stream
+     */
+    @Timed(SEARCH_CALL_LATENCY_TIMER)
+    public SearchResults getAllIssues(StringBuilder jql, int startAt,
+                                      JiraSourceConfig configuration) {
+
+        String url = configuration.getAccountUrl() + REST_API_SEARCH;
+        if (configuration.getAuthType().equals(OAUTH2)) {
+            url = authConfig.getUrl() + REST_API_SEARCH;
+        }
+
+        URI uri = UriComponentsBuilder.fromHttpUrl(url)
+                .queryParam(MAX_RESULT, FIFTY)
+                .queryParam(START_AT, startAt)
+                .queryParam(JQL_FIELD, jql)
+                .queryParam(EXPAND_FIELD, EXPAND_VALUE)
+                .buildAndExpand().toUri();
+        return invokeRestApi(uri, SearchResults.class).getBody();
+    }
+
+    /**
+     * Gets issue.
+     *
+     * @param issueKey the item info
+     * @return the issue
+     */
+    @Timed(TICKET_FETCH_LATENCY_TIMER)
+    public String getIssue(String issueKey) {
```

Comment:
Create a class that abstracts calls.

I'd actually suggest calling it `JiraClient`, but that is already used. Maybe `JiraRestClient` instead.

All of the calls to Jira should go through there. You should be able to remove `RestTemplate` from this class and have it only in `JiraRestClient`.

```
class JiraRestClient {
  Issue getIssue(String issueKey) {
    ...
  }
  public SearchResults getAllIssues(StringBuilder jql, int startAt,
                                      JiraSourceConfig configuration) {
...
```

Reply:
Introduced `JiraRestClient` 👍